puppet-mode.el

```diff
@@ -1093,7 +1128,11 @@ for each entry."
   (setq-local beginning-of-defun-function #'puppet-beginning-of-defun-function)
   ;; Indentation
   (setq-local indent-line-function #'puppet-indent-line)
-  (setq indent-tabs-mode puppet-indent-tabs-mode)
+  (if puppet-use-smie
+      (smie-setup puppet-smie-grammar #'puppet-smie-rules
+                  :forward-token #'puppet--smie-forward-token
+                  :backward-token #'puppet--smie-backward-token)
+      (setq indent-tabs-mode puppet-indent-tabs-mode))
```

Comment:
I see, but that was likely a mistake I made back then.  I think you can move it out of `if`, and replace the `if` with a `when`.